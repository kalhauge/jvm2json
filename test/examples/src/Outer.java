public class Outer <T> {
    public class Inner1 <K> { }
    public class Inner2 extends Inner1<java.lang.String> { }
}
