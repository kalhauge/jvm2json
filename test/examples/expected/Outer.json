{"name":"Outer","access":["public","super"],"typeparams":[{"name":"T","classbound":{"kind":"class","name":"java/lang/Object","inner":null,"args":[],"annotations":[]},"interfacebound":[],"annotations":[]}],"super":{"name":"java/lang/Object","inner":null,"args":[],"annotations":[]},"interfaces":[],"fields":[],"methods":[{"name":"<init>","access":["public"],"typeparams":[],"params":[],"returns":{"type":null,"annotations":[]},"code":{"max_stack":1,"max_locals":1,"exceptions":[],"stack_map":null,"bytecode":[{"offset":0,"opr":"load","type":"ref","index":0},{"offset":1,"opr":"invoke","access":"special","method":{"is_interface":false,"ref":{"kind":"class","name":"java/lang/Object"},"name":"<init>","args":[],"returns":null}},{"offset":4,"opr":"return","type":null}]},"annotations":[],"exceptions":[],"default":null}],"bootstrapmethods":[],"enclosingmethod":null,"innerclasses":[{"class":"Outer$Inner2","outer":"Outer","name":"Inner2","access":["public"]},{"class":"Outer$Inner1","outer":"Outer","name":"Inner1","access":["public"]}],"annotations":[],"version":[63,0]}